### ISSUE TITLE
Add keyboard navigation to search

### ISSUE THREAD

**Issue by atomtay** [MAPC/datacommon#373]
**Describe the feature. What should a user be able to do?**
After typing in a search string in the home search bar, the user should be able to use the keyboard (either through tab or arrow keys) to access a search result.

**Why do we need this feature?**
I always forget that you can't use the arrow keys to select an option, which is a minor annoyance. More notably, though, without any method of keyboard navigation, our search feature is inaccessible to users who do not use a mouse.

**Additional context**
This feature could be implemented alongside #371 or #372 in a re-implementation of DataCommon search.
